### DISCUSSION TITLE
+ HTTP/3 support

### DISCUSSION THREAD

**Discussion by tomByrer** [vlang/v#11875] [Ideas]
Backed by UDP/Quic, is thought of being more secure than HTTP/* & even faster.  Could also save RAM & CPU due to fewer ACK round trips.
[Browser support](https://caniuse.com/http3) is very high, with only Safari not automatically turned on, but still available for testing & power-users behind a flag.

Some research for HTTP/3 in [JustJS is here](https://github.com/just-js/just/issues/14).

**Comment by bstnbuck** [comment]
Hi, I think both http2 and http3 are planned and will be implemented earlier or later.
Both http2 and QUIC (which is needed  for http3) need TLS encryption (QUIC only uses TLS 1.3). Since TLS is not yet implemented in V, such an implementation cannot be thought of. 
Reference: #2066, #10386

**Reply by blackshirt** [reply]
> Since TLS is not yet implemented in V, such an implementation cannot be thought of.
> Reference: #2066, #10386

Actually, I'm playing a bit with (subset) TLS 1.3 only in pure v, but its working in preliminary testing ... you can look at [tls 1.3](https://github.com/blackshirt/tls13)
Its very basic thing,..currently, there are many component not implemented, and need more building block for this, like asn.1 and, certificate handling. If we can make it happen, we have building block for building QUIC or HTTP/3 in pure v

**Comment by tomByrer** [comment]
TY for the insights.
Ref 2066 is tagged "Status: After 1.0".  Which means in about 8 years?  (Just getting to v0.2 took much longer than promised.)
I can't wait that long.

**Comment by dumblob** [comment]
@tomByrer just use https://github.com/h2o/h2o as a [library](https://github.com/h2o/h2o/issues/2611 ) from *V* and you should be good to go right now.

**Reply by Jengro777** [reply]
Great answer, even better if there are use cases available

**Comment by tomByrer** [comment]
Oh wow thanks @dumblob!  Nice find & good advice.

**Comment by ouvaa** [comment]
@tomByrer @dumblob can any of you show an example of v using these libraries? with pico parser? or the fastest https possible to have the fastest performance

**Reply by dumblob** [reply]
Hi @ouvaa, https is quite independent from the underlying HTTP protocol version. Fastest https then might translate to "fastest SSL processing" which has nothing to do with the *pico parser* nor the *h2o* library I linked above.

Please look for benchmarks of SSL libraries (there are plenty) if you are after "fastest https".

**Reply by ouvaa** [reply]
@dumblob totally "new" to vlang, been touching it 2 years ago but realised missing lots of lib so went back to golang. now back coz i'm truly looking for the fastest https etc.

possible to show how to implement the h2o http2/3 using the pico.v example? totally clueless how u guys made it sound so simple and easy to do.

https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/V/pico.v/main.v

**Comment by Jengro777** [comment]
How does veb write http3? 
I don't seem to see the case.

**Reply by medvednikov** [reply]
we need http2 first

**Reply by Jengro777** [reply]
> we need http2 first

Yes, one at a time.

**Comment by Jengro777** [comment]
Should say http3 support, put in the road map
